### PR TITLE
Extend device_devices table

### PR DESCRIPTION
**Pull request by S7evinK**
### Pull Request Checklist

Adds the columns `last_seen_ts`, `ip` and `user_agent`. At the moment they are only set when registering a new account.

* [ ] I have added any new tests that need to pass to `sytest-whitelist` as specified in [docs/sytest.md](https://github.com/matrix-org/dendrite/blob/master/docs/sytest.md)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/master/docs/CONTRIBUTING.md#sign-off)

### CODE REVIEW

File: userapi/storage/devices/postgres/deltas/20201001204705_last_seen_ts_ip.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+-- +goose Up
+-- +goose StatementBegin
+ALTER TABLE device_devices ADD COLUMN last_seen_ts BIGINT NOT NULL;
+ALTER TABLE device_devices ADD COLUMN ip TEXT;
+ALTER TABLE device_devices ADD COLUMN user_agent TEXT;
```
Comment (Kegsay) [inline userapi/storage/devices/postgres/deltas/20201001204705_last_seen_ts_ip.sql:5]: These will need IF NOT EXISTS

**Review by neilalexander — APPROVED** [review]

File: userapi/storage/devices/sqlite3/storage.go [hunk 2]
```diff
@@ -193,3 +193,10 @@ func (d *Database) RemoveAllDevices(
 	})
 	return
 }
+
+// UpdateDeviceLastSeen updates a the last seen timestamp and the ip address
+func (d *Database) UpdateDeviceLastSeen(ctx context.Context, deviceID, ipAddr string) error {
+	return sqlutil.WithTransaction(d.db, func(txn *sql.Tx) error {
```
Comment (Kegsay) [inline userapi/storage/devices/sqlite3/storage.go:199]: This should probably use a txn writer

**Review by Kegsay — APPROVED** [review]
LGTM aside from comments!

File: userapi/storage/devices/postgres/deltas/20201001204705_last_seen_ts_ip.sql (hunk 1 above)
Comment (S7evinK) [reply userapi/storage/devices/postgres/deltas/20201001204705_last_seen_ts_ip.sql:5]: No idea how i missed that.

**Pull request merged**